lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -193,6 +200,53 @@ defmodule Mix.Compilers.Elixir do
     Mix.shell.info "Compiling #{source} (it's taking more than #{threshold}s)"
   end
 
+  defp warn_for_missing_remote_functions(runtime_remote_calls, source, :module),
+    do: do_warn_for_missing_remote_functions(runtime_remote_calls, source)
+
+  defp warn_for_missing_remote_functions(runtime_remote_calls, source, {:impl, _}),
+    do: do_warn_for_missing_remote_functions(runtime_remote_calls, source)
+
+  defp warn_for_missing_remote_functions(_, _, _),
+    do: :ok
+
+  defp do_warn_for_missing_remote_functions(runtime_remote_calls, source) do
+    Enum.sort_by(runtime_remote_calls, fn {_, _, line} -> line end)
+    |> Enum.each(fn {module, {func, arity} = fas, line} ->
+      if Code.ensure_loaded?(module) do
+        unless fas in exports_for_module(module) do
+          IO.warn(
+            "Remote function #{inspect module}.#{func}/#{arity} cannot be found\n" <>
+            "  #{source}:#{line}",
+            []
+          )
+        end
+      else
+        IO.warn(
+          "Module #{inspect module} cannot be found\n" <>
+          "  In remote call to #{inspect module}.#{func}/#{arity} at:\n" <>
+          "    #{source}:#{line}",
+          []
+        )
+      end
+    end)
+  end
+
+  defp exports_for_module(module) do
+    if match?("Elixir." <> _, to_string(module)) do
```

Comment:
Don't match on the module name, check for `function_exported?(module, :__info__, 1)`.